hedera-node/docs/design/services/smart-contract-service/accounts-blocklisting.md

```diff
@@ -0,0 +1,50 @@
+# Accounts Blocklisting
+
+## Purpose
+
+In Ethereum, everyone that knows a private key is able to authorize a transaction from the corresponding EVM public address, and the first to send a transaction would be eligible to withdraw any funds from that address.
+In Hedera, it is possible for an account with associated EVM address alias to have a key that does not correspond to the alias. This is what we refer to as “EVM address squatting” i.e. using a key that’s different from the key that’s used to derive the EVM address alias from.
+
+It is possible to “squat” on an EVM address in a couple of ways:
+1. Create an account with an alias and then update the account key.
+2. Create an account with an alias and a key that’s different from the key that’s used to derive the alias from.
+
+In both cases, this results in the private key losing control over the EVM address derived from the corresponding public key.
+
+This means that there is a difference in how some situations, like accidental transfers to known EVM addresses, will be handled in Ethereum and in Hedera:
+- In Ethereum, everyone that knows the corresponding private key will be able to authorize a transaction from the known address, and the first to send a transaction would be eligible to withdraw the funds.
+- In Hedera, only the account with the associated corresponding alias would be eligible to withdraw the funds. If another key controls this account, then only that key can be used to withdraw the funds.
+
+## Goals
+
+- We want to block a list of known addresses, so that no one can “squat” on them and exercise full control on any accidentally sent and blocked funds. Sending funds to such blocked addresses should not be successful as these accounts are considered “compromised”.
+
+## Non Goals
+
+- We don't consider unblocking already blocked accounts.
+- We don't consider what happens to a blocked account when it expires.
+
+## Architecture
+
+The implementation related to blocked accounts will be gated by a feature flag.
```

Comment:
Maybe we should list the feature flags and their effect, the structure of the file used to list all the private keys

Reply:
There is only one feature flag related to this, do you mean to also add the property that has the name of the resource file that contains the blocklist as well?

As for the structure of the file, there is a `File containing the list of blocked accounts` two lines below this which describes that, do you mean to update it in some way?

Reply:
Somehow I missed the next section when I wrote this, my bad 😊